Adding method for progress updates on the NVC delegate. (#2224)
mapbox/mapbox-navigation-ios: CHANGELOG.md

## master
* Fixed an issue where swiping the banner down after the StepsTableViewController has already displayed could put the UI in an unstable state. ([#2197](https://github.com/mapbox/mapbox-navigation-ios/pull/2197))

> * Added `NavigationViewControllerDelegate.navigationViewController(_:didUpdate:location:rawLocation)` method that allows the client developer to capture progress updates without having to inject themselves between the `NavigationViewController` and the `NavigationService`.  ([#2224](https://github.com/mapbox/mapbox-navigation-ios/pull/2224))

Reviewer: Does this work?

```suggestion
* Added `NavigationViewControllerDelegate.navigationViewController(_:didUpdate:location:rawLocation)` method that allows you to capture progress updates without having to inject a class between the `NavigationViewController` and the `NavigationService`.  ([#2224](https://github.com/mapbox/mapbox-navigation-ios/pull/2224))
```

> * Added `NavigationViewControllerDelegate.navigationViewController(_:didUpdate:location:rawLocation)` method that allows the client developer to capture progress updates without having to inject themselves between the `NavigationViewController` and the `NavigationService`.  ([#2224](https://github.com/mapbox/mapbox-navigation-ios/pull/2224))

Reviewer 2: i wonder what if we decide to cherry-pick this commit at some point to the patch release branch? would coupling changelog record with the actual fix be a limitation?

Reviewer: If we were to cherry-pick this change, we’d most likely revise the changelog entry on both branches to match.

There has been a lot of discussion about how to manage changelogs in the map SDK projects. The main benefit to including the changlog blurb in the change is that we don’t have to go back and try to remember what this PR was about at release time. (In the old days, we would try to put together a Paper document, but it easily fell out of sync and took extra time at release.)

Fortunately, we’ve only had occasional need for one-off patch branches so far, so it hasn’t posed a problem for us. Over time, if the bandwidth of development picks up to the point where we need to maintain long-running release branches, we could consider a different strategy.
